### PR TITLE
[master] Fix _find_libcrypto for MacOS and Onedir

### LINKED ISSUE #64877: [BUG] Mac Onedir Cannot locate OpenSSL libcrypto
**Description**
When running salt-commands from an unpacked Salt onedir package on MacOS the included libcrypto libraries are not available because of code that looks for them in the path where Salt is usually installed on MacOS, eg. /opt/salt/

**Setup**
Download Mac onedir tar file, unpack, and attempt to run a Salt command.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Look in location salt is running from, this accounts for running from an unpacked onedir file that has not been installed.

### What issues does this PR fix or reference?
Fixes: #64877 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by whytewolf — DISMISSED** [review]

File: salt/utils/rsax931.py [hunk 1]
```diff
@@ -37,6 +37,12 @@ def _find_libcrypto():
         # system.
         # look in salts pkg install location.
         lib = glob.glob("/opt/salt/lib/libcrypto.dylib")
+
+        # look in location salt is running from
+        # this accounts for running from an unpacked
+        # onedir file
+        lib = lib or glob.glob(os.path.join("lib/libcrypto.dylib"))
```
Comment (s0undt3ch) [inline salt/utils/rsax931.py:44]:
```suggestion
        lib = lib or glob.glob("lib/libcrypto.dylib")
```
No need for `os.path.join` right?

File: tests/unit/utils/test_rsax931.py [hunk 2]
```diff
@@ -276,3 +276,30 @@ def test_load_libcrypto(self):
             or hasattr(lib, "OPENSSL_init_crypto")
             or hasattr(lib, "OPENSSL_no_config")
         )
+
+    @patch.object(salt.utils.platform, "is_darwin", lambda: True)
```
Comment (s0undt3ch) [inline tests/unit/utils/test_rsax931.py:280]: Let's move all these patch decorators to the test function both using `with` contexts please.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
This should be going into 3006.x too correct?

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/unit/utils/test_rsax931.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+import glob
+import os
+import platform
+import sys
+
+import pytest
+
+import salt.utils.platform
+
+# salt libs
```
Comment (s0undt3ch) [inline tests/pytests/unit/utils/test_rsax931.py:17]:
```suggestion
```

**Review by s0undt3ch — DISMISSED** [review]
Lets just remove these unnecessary comments.

**Review by s0undt3ch — APPROVED** [review]

File: salt/utils/rsax931.py [hunk 4]
```diff
@@ -37,6 +37,12 @@ def _find_libcrypto():
         # system.
         # look in salts pkg install location.
         lib = glob.glob("/opt/salt/lib/libcrypto.dylib")
+
+        # look in location salt is running from
+        # this accounts for running from an unpacked
+        # onedir file
+        lib = lib or glob.glob("lib/libcrypto.dylib")
```
Comment (dwoz) [inline salt/utils/rsax931.py:44]: Can't this just be a `find_library('libcrypto')` call?

**Review by Ch3LL — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**